src/callbacks_step/timeseries.jl

```diff
@@ -0,0 +1,202 @@
+
+"""
+    TimeseriesCallback(semi, point_coordinates;
+                       interval=1, solution_variables=cons2cons,
+                       output_directory="out", filename="timeseries.h5",
+                       RealT=real(solver), uEltype=eltype(cache.elements))
+
+Create a callback that records point-wise data at points given in `point_coordinates` every
+`interval` time steps. By default, the conservative variables are recorded, but this can be
+controlled by passing a different conversion function to `solution_variables`.
+
+After the last time step, the results are stored in an HDF5 file `filename` in directory
+`output_directory`.
+
+The real data type `RealT` and data type for solution variables `uEltype` default to the respective
+types used in the solver and the cache.
+
+!!! warning "Experimental implementation"
+    This is an experimental feature and may change in future releases.
+"""
+mutable struct TimeseriesCallback{RealT<:Real, uEltype<:Real, SolutionVariables, VariableNames, Cache}
+  interval::Int
+  solution_variables::SolutionVariables
+  variable_names::VariableNames
+  output_directory::String
+  filename::String
+  point_coordinates::Matrix{RealT}
+  point_data::Vector{Vector{uEltype}}
```

Comment:
Could you please comment here on the memory layout? Is it something like
- outer vector for each point where you want to record the solution
- inner vector for all single-point variables and times

Reply:
Done in 407632c8. Please let me know if this is sufficient.